authentication/auth.go

```diff
@@ -402,15 +406,15 @@ func TransformSchemaParamsToName(postValues url.Values) string {
 }
 
 // GenerateTokenFromPost converts a set of POST values into a JWT
-func GenerateTokenFromPost(postValues url.Values) (string, string) {
+func GenerateTokenFromPost(postValues url.Values, launchV2 bool) (string, string) {
```

Comment:
- The actual token structure need updating to conform to the v2 data format as documented. All this is doing is sending the `version` key in the URL which is just 1 of the difference. 
- The actual structure of the payload is different between v1/v2 which launcher needs to handle based on the version we want to launch. Consult the docs for the differences and the struct itself. For example v1 would be a flat key value pair, whereas for v2, the metadata is nested.

Reply:
Updated the structures for v1/v2 and business/ad-hoc.